parcels/include/parcels.h

```diff
@@ -954,6 +954,265 @@ static inline StatusCode temporal_interpolationUVW_c_grid(type_coord x, type_coo
+            f_w = f_w * (1 - .5 * eta) / (1 - eta);
+          }
+        } else if (interp_method == FREESLIP) {
+          f_u = f_u / (1 - eta);
+          if (withW) {
+            f_w = f_w / (1 - eta);
+          }
+        }
+      }
+      if ((is_zero_flt(dataU[0][0][0])) && (is_zero_flt(dataU[0][1][0])) && (is_zero_flt(dataU[1][0][0])) && (is_zero_flt(dataU[1][1][0])) &&
+          (is_zero_flt(dataV[0][0][0])) && (is_zero_flt(dataV[0][1][0])) && (is_zero_flt(dataV[1][0][0])) && (is_zero_flt(dataV[1][1][0])) &&
+          xsi > 0.){
+        if (interp_method == PARTIALSLIP) {
+          f_v = f_v * (.5 + .5 * xsi) / xsi;
+          if (withW) {
+            f_w = f_w * (.5 + .5 * xsi) / xsi;
+          }
+        } else if (interp_method == FREESLIP) {
+          f_v = f_v / xsi;
+          if (withW) {
+            f_w = f_w / xsi;
+          }
+        }
+      }
+      if ((is_zero_flt(dataU[0][0][1])) && (is_zero_flt(dataU[0][1][1])) && (is_zero_flt(dataU[1][0][1])) && (is_zero_flt(dataU[1][1][1])) &&
+          (is_zero_flt(dataV[0][0][1])) && (is_zero_flt(dataV[0][1][1])) && (is_zero_flt(dataV[1][0][1])) && (is_zero_flt(dataV[1][1][1])) &&
+          xsi < 1.){
+        if (interp_method == PARTIALSLIP) {
+          f_v = f_v * (1 - .5 * xsi) / (1 - xsi);
+          if (withW) {
+            f_w = f_w * (1 - .5 * xsi) / (1 - xsi);
+          }
+        } else if (interp_method == FREESLIP) {
+          f_v = f_v / (1 - xsi);
+          if (withW) {
+            f_w = f_w / (1 - xsi);
+          }
+        }
+      }
+      if ((is_zero_flt(dataU[0][0][0])) && (is_zero_flt(dataU[0][0][1])) && (is_zero_flt(dataU[0][1][0])) && (is_zero_flt(dataU[0][1][1])) &&
+          (is_zero_flt(dataV[0][0][0])) && (is_zero_flt(dataV[0][0][1])) && (is_zero_flt(dataV[0][1][0])) && (is_zero_flt(dataV[0][1][1])) &&
+          zeta > 0.){
+        if (interp_method == PARTIALSLIP) {
+          f_u = f_u * (.5 + .5 * zeta) / zeta;
+          f_v = f_v * (.5 + .5 * zeta) / zeta;
+        } else if (interp_method == FREESLIP) {
+          f_u = f_u / zeta;
+          f_v = f_v / zeta;
+        }
+      }
+      if ((is_zero_flt(dataU[1][0][0])) && (is_zero_flt(dataU[1][0][1])) && (is_zero_flt(dataU[1][1][0])) && (is_zero_flt(dataU[1][1][1])) &&
+          (is_zero_flt(dataV[1][0][0])) && (is_zero_flt(dataV[1][0][1])) && (is_zero_flt(dataV[1][1][0])) && (is_zero_flt(dataV[1][1][1])) &&
+          zeta < 1.){
+        if (interp_method == PARTIALSLIP) {
+          f_u = f_u * (1 - .5 * zeta) / (1 - zeta);
+          f_v = f_v * (1 - .5 * zeta) / (1 - zeta);
+        } else if (interp_method == FREESLIP) {
+          f_u = f_u / (1 - zeta);
+          f_v = f_v / (1 - zeta);
+        }
+      }
+      *u *= f_u;
+      *v *= f_v;
+      if (withW) {
+        *w *= f_w;
+      }
+
+  return SUCCESS;
+}
+
+static inline StatusCode temporal_interpolation_slip(type_coord x, type_coord y, type_coord z, double time, CField *U, CField *V, CField *W,
```

Comment:
function looks very neat and tidy - nice.

Reply:
Thanks!